paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -182,19 +183,28 @@ def get_http_utilization_for_all_tasks(marathon_service_config, marathon_tasks,
 
     endpoint = endpoint.lstrip('/')
     utilization = []
-    for task in marathon_tasks:
-        try:
-            utilization.append(json_mapper(get_json_body_from_service(task.host, task.ports[0], endpoint)))
-        except requests.exceptions.Timeout:
+
+    monkey.patch_socket()
+    gevent_pool = pool.Pool(20)
+    jobs = [gevent_pool.spawn(json_mapper, get_json_body_from_service(task.host, task.ports[0], endpoint))
+            for task in marathon_tasks]
+    gevent.joinall(jobs)
+
+    for job in jobs:
+        if job.value is not None:
+            utilization.append(job.value)
+        elif isinstance(job.exception, requests.exceptions.Timeout):
             # If we time out querying an endpoint, assume the task is fully loaded
             # This won't trigger in the event of DNS error or when a request is refused
             # a requests.exception.ConnectionError is raised in those cases
```

Comment:
There's a subtle change here that means this comment isn't true now if I'm reading it right?

i.e. now if there's any exception querying a task we assume it to be "fully loaded". That might actually be reasonable although I think just dropping it from the list is also reasonable given that we are taking an average.

I guess I'm happy either way... but please update the comment if you want to keep the code like this.